libs/simdvec/src/main/java/org/elasticsearch/simdvec/ES92Int7VectorsScorer.java

```diff
@@ -122,12 +122,37 @@ public void scoreBulk(
         float centroidDp,
         float[] scores
     ) throws IOException {
-        int7DotProductBulk(q, BULK_SIZE, scores);
-        in.readFloats(lowerIntervals, 0, BULK_SIZE);
-        in.readFloats(upperIntervals, 0, BULK_SIZE);
-        in.readInts(targetComponentSums, 0, BULK_SIZE);
-        in.readFloats(additionalCorrections, 0, BULK_SIZE);
-        for (int i = 0; i < BULK_SIZE; i++) {
+        scoreBulk(
+            q,
+            queryLowerInterval,
+            queryUpperInterval,
+            queryComponentSum,
+            queryAdditionalCorrection,
+            similarityFunction,
+            centroidDp,
+            scores,
+            BULK_SIZE
+        );
+    }
+
+    public void scoreBulk(
```

Comment:
I am wondering if this method needs to be implemented in the subclasses in order to take advantage of the optimizations.